src/utils.ts

```diff
@@ -158,4 +159,67 @@ export function createDidPayload(verificationMethods: VerificationMethod[], veri
             authentication: verificationKeys.map(key => key.keyId)
         }
     )
+}
+
+export function createDidPayloadWithSignInputs(seed?: string, keys?: IKeyPair[]) {
+    if(seed && keys) throw new Error ('Only one of seed or keys should be passed as an argument')
+    
+    if(!keys) {
+        keys = [seed ? createKeyPairBase64(seed) : createKeyPairBase64()]
+    }
+
+    const verificationMethodTypes = keys.map((key) => !key.algo || key.algo == MethodSpecificIdAlgo.Base58 ? VerificationMethods.Base58 : VerificationMethods.JWK)
+    const verificationKeys = keys.map((key, i) => createVerificationKeys(key, key.algo || MethodSpecificIdAlgo.Base58, `key-${i}`))
+    const verificationMethod = createDidVerificationMethod(verificationMethodTypes, verificationKeys)
+    
+    let payload : Partial<MsgCreateDidPayload> = {
+        id: verificationKeys[0].didUrl,
+        controller: verificationKeys.map(key => key.didUrl),
+        verificationMethod: verificationMethod,
+        authentication: verificationKeys.map(key => key.keyId),
+    }
+
+    const keyHexs = keys.map((key)=>convertKeyPairtoTImportableEd25519Key(key))
+    const signInputs = keyHexs.map((key)=>createSignInputsFromImportableEd25519Key(key, verificationMethod))
```

Comment:
I'd suggest to simplify this by creating sign input pairs directly rather than doing a 2-way conversion.

Reply:
I have to create a new createSignInputsFromKeypair function?